{{cookiecutter.github_repository}}/{{cookiecutter.main_module}}/users/auth/serializers.py

```diff
@@ -43,3 +45,70 @@ def get_name(self, obj):
 
     def get_auth_token(self, obj):
         return get_token_for_user(obj, "authentication")
+
+
+class PasswordChangeSerializer(serializers.Serializer):
+    current_password = serializers.CharField(required=True)
+    new_password = serializers.CharField(required=True)
+
+    error_messages = {
+        'invalid_password': 'Invalid password.'
+    }
+
+    def validate_current_password(self, value):
+        if not self.context['request'].user.check_password(value):
+            raise serializers.ValidationError(self.error_messages['invalid_password'])
+        return value
+
+    def validate_new_password(self, value):
+        # https://docs.djangoproject.com/en/2.0/topics/auth/passwords/#django.contrib.auth.password_validation.validate_password
+        password_validation.validate_password(value)
+        return value
+
+
+class PasswordResetSerializer(serializers.Serializer):
+    email = serializers.EmailField(required=True)
+
+    error_messages = {
+        'email_does_not_exist': "No user with the specified email found"
+    }
+
+    def validate_email(self, value):
+        user = User.objects.filter(email__iexact=value, is_active=True).first()
```

Comment:
What's the thought behind have `is_active=True`. This means that inactive user's won't be able to change the password? 

Let's not treat the is_active as is_deleted. The message is saying email doesn't exist, but that's not the case. `is_active` can have different meanings in different cases, but not allowing password reset if the user is not active, doesn't make much sense to me.

Reply:
Yes, that was the thought. Inactive users should not be able to reset their passwords.
I will remove the condition.